### ISSUE TITLE
Add dodging to violin plots that call :color 

### ISSUE THREAD

**Issue by hbsmith** [GiovineItalia/Gadfly.jl#1635]
I'm trying to make violin plot versions of the boxplots which are displayed as p2 here: https://gadflyjl.org/stable/gallery/geometries/#[Geom.boxplot](@ref)
<img width="398" alt="image" src="https://github.com/user-attachments/assets/e801c3d5-dafe-4733-bc20-b5279f83ebe2">

My expectation is that `:color` in a violin plot would would like it does in a boxplot, but it does not. If I want to plot two different categories using a violin plot, where both categories have the same x-axis, it does not work. I'd expect it to apply dodge to the datasets. After digging, it seems like there is no option to even have violin plot dodge. 

Basically, I want to make the same kind of plot that seaborn demonstrates here: https://seaborn.pydata.org/generated/seaborn.violinplot.html
`sns.violinplot(data=df, x="class", y="age", hue="alive")`
![image](https://github.com/user-attachments/assets/2945c81b-bb7e-430e-bd2b-ca3175da6286)

Is there a way to do this that I'm not aware of?

**Comment by Mattriks** [comment]
Yes the dodge effect isn't implemented yet for `Geom.violin`, and we're happy to see PRs!  A possible workaound is:
```julia
plot(salaries, x=:Rank, y=:Salary, color=:Rank,
    xgroup=:Discipline,
    Geom.subplot_grid(Geom.violin),
    Scale.xgroup(levels=["Discipline A", "Discipline B"]),
    Guide.xlabel(nothing)
)
```
